src/Chunk.cpp

```diff
@@ -1694,19 +1700,11 @@ cBlockEntity * cChunk::GetBlockEntity(int a_BlockX, int a_BlockY, int a_BlockZ)
 	ASSERT(a_BlockZ >= m_PosZ * cChunkDef::Width);
 	ASSERT(a_BlockZ < m_PosZ * cChunkDef::Width + cChunkDef::Width);
 
-	for (cBlockEntityList::iterator itr = m_BlockEntities.begin(); itr != m_BlockEntities.end(); ++itr)
-	{
-		if (
-			((*itr)->GetPosX() == a_BlockX) &&
-			((*itr)->GetPosY() == a_BlockY) &&
-			((*itr)->GetPosZ() == a_BlockZ)
-		)
-		{
-			return *itr;
-		}
-	}  // for itr - m_BlockEntities[]
+	int RelX = a_BlockX - m_PosX * cChunkDef::Width;
+	int RelZ = a_BlockZ - m_PosZ * cChunkDef::Width;
 
-	return nullptr;
+	auto itr = m_BlockEntities.find(MakeIndex(RelX, a_BlockY, RelZ));
+	return itr == m_BlockEntities.end() ? nullptr : itr->second;
```

Comment:
Style: parentheses around comparison.
(I don't want to have to look up whether this is evaluated as `(a == b) ? c : d` or `a == (b ? c : d)` )